project/Common.scala

```diff
@@ -66,17 +66,31 @@ object Common extends AutoPlugin {
           version.value,
           "-sourcepath",
           (baseDirectory in ThisBuild).value.toString,
-          "-doc-source-url", {
-            val branch = if (isSnapshot.value) "master" else s"v${version.value}"
-            s"https://github.com/akka/alpakka/tree/${branch}€{FILE_PATH}.scala#L1"
-          },
           "-skip-packages",
           "akka.pattern:" + // for some reason Scaladoc creates this
           "org.mongodb.scala:" + // this one is a mystery as well
           // excluding generated grpc classes, except the model ones (com.google.pubsub)
           "com.google.api:com.google.cloud:com.google.iam:com.google.logging:" +
           "com.google.longrunning:com.google.protobuf:com.google.rpc:com.google.type"
         ),
+      Compile / doc / scalacOptions ++= (scalaVersion.value match {
+          case Dependencies.Scala211 =>
+            Seq(
+              "-doc-source-url", {
+                val branch = if (isSnapshot.value) "master" else s"v${version.value}"
+                s"https://github.com/akka/alpakka/tree/${branch}€{FILE_PATH}.scala#L1"
+              }
+            )
```

Comment:
That's right. Even if we generate the unified scaladoc only with Scala 2.12, the scaladoc in the javadoc jars is generated with whatever Scala version is used to compile the code.